tools/generators/pbxproject_prefix/src/Generator/BuildSettings.swift

```diff
@@ -0,0 +1,119 @@
+import GeneratorCommon
+import PBXProj
+
+extension Generator {
+    /// Calculates the `buildSettings` attribute of the `XCBuildConfiguration`
+    /// elements used by the `PBXProject` element.
+    ///
+    /// - Parameters:
+    ///   - buildMode: The `BuildMode`.
+    ///   - indexingProjectDir: The value returned from
+    ///     `Generator.indexingProjectDir()`.
+    ///   - workspace: The absolute path to the Bazel workspace.
+    static func buildSettings(
+        buildMode: BuildMode,
+        indexingProjectDir: String,
+        workspace: String
+    ) -> String {
+        var settings: [String: String] = [
+            "ALWAYS_SEARCH_USER_PATHS": "NO",
+            "BAZEL_CONFIG": "rules_xcodeproj",
+            "BAZEL_EXTERNAL": "$(BAZEL_OUTPUT_BASE)/external".pbxProjEscaped,
```

Comment:
I could escape all of these static strings manually, but I found it more readable this way.